File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(numbers, target, idx, ref):
        if idx == len(numbers):
            if ref == target:
                nonlocal answer
                answer += 1
        else:
            dfs(numbers, target, idx+1, ref + numbers[idx])
            dfs(numbers, target, idx+1, ref - numbers[idx])

    dfs(numbers, target, 0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    public int solution(int[] numbers, int target) {
        
        findTarget(numbers, target, 0, 0);
        return answer;
    }
    
    public void findTarget(int[] numbers, int target, int sum, int index){
        if(index == numbers.length){
            if(sum == target){
                answer ++;
            }
        }
        else{
            findTarget(numbers, target, sum+numbers[index], index+1);
            findTarget(numbers, target, sum-numbers[index], index+1);
        }
    }
}